test/spec/geo/leaflet/leaflet-cartodb-webgl-layer-group-view.spec.js

```diff
@@ -0,0 +1,60 @@
+var L = require('leaflet');
+var VisModel = require('../../../../src/vis/vis');
+var LayersCollection = require('../../../../src/geo/map/layers');
+var CartoDBLayer = require('../../../../src/geo/map/cartodb-layer');
+var CartoDBLayerGroup = require('../../../../src/geo/cartodb-layer-group');
+var LeafletCartoDBWebGLLayerGroupView = require('../../../../src/geo/leaflet/leaflet-cartodb-webgl-layer-group-view');
+
+describe('src/geo/leaflet/leaflet-cartodb-webgl-layer-group-view.js', function () {
+  beforeEach(function () {
+    var container = document.createElement('div');
+    container.id = 'map';
+    document.body.appendChild(container);
+
+    this.leafletMap = L.map('map');
+    this.layersCollection = new LayersCollection();
+    this.layerGroupModel = new CartoDBLayerGroup({}, {
+      layersCollection: this.layersCollection
+    });
+    this.layerGroupModel.getTileURLTemplates = function () { return [ 'http://carto.com/{z}/{x}/{y}.png' ]; };
+
+    this.vis = new VisModel();
+
+    this.layer1 = new CartoDBLayer({ id: 'layer1' }, { vis: this.vis });
+    this.layer2 = new CartoDBLayer({ id: 'layer2' }, { vis: this.vis });
+
+    this.layersCollection.reset([ this.layer1, this.layer2 ]);
+  });
+
+  afterEach(function () {
+    document.getElementById('map').remove();
+  });
+
+  it('should set the styles options when meta changes', function () {
+    var view = new LeafletCartoDBWebGLLayerGroupView(this.layerGroupModel, this.leafletMap); // eslint-disable-line
+    var self = this;
+
+    setTimeout(function () {
+      self.layer1.set('meta', {cartocss: '#layer {polygon-fill: red;}'});
+      self.layer2.set('meta', {cartocss: '#layer {polygon-fill: blue;}'});
+
+      expect(view.options.styles).toEqual([ '#layer {polygon-fill: red;}', '#layer {polygon-fill: blue;}' ]);
+      done(); // eslint-disable-line
+    }, 100);
+  });
+
+  it('should set the visibility options when visibility changes', function () {
+    var view = new LeafletCartoDBWebGLLayerGroupView(this.layerGroupModel, this.leafletMap); // eslint-disable-line
+    var self = this;
+
+    setTimeout(function () {
+      self.layer1.set('visible', true);
+      self.layer2.set('visible', false);
+
+      expect(view.tangram.scene.config.layers.layer1.visible).toBeTruthy();
+      expect(view.tangram.scene.config.layers.layer1.visible).toBeFalsy();
+
+      done(); // eslint-disable-line
+    }, 100);
+  });
```

Comment:
Here's an idea to reduce the number of details these tests know about tangram (eg: `view.tangram.scene.config.layers.layer1.visible`):

```
// code -> we extract a function that initialises tangram, that we'll then override in our tests.

var LeafletCartoDBWebglLayerGroupView = L.Class.extend({
  initialize: function (layerGroupModel, map) {

    // ...
    this.tangram = this._initializeTangram(map);

    // ...
  },

  _initializeTangram: function (map) {
    return new TC(map);
  },

  // ...
}

// test

// we declare a fake Tangram with spies
var FakeTangram = function () {};
FakeTangram.prototype = {
  addLayer: jasmine.createSpy('addLayer'),
  addDataSource: jasmine.createSpy('addDataSource')
}

// We extend our original class for this test, only so that it uses our fake Tangram
var LeafletCartoDBWebglLayerGroupView = LeafletCartoDBWebglLayerGroupView.extend({
  _initializeTangram: function (map) {
    return new FakeTangram(map);
  }
});

// We use the spies to ensure LeafletCartoDBWebglLayerGroupView is sending the right messages to Tangram 
it('should add a layer to tangram when something happens', function () {
  // do something

  expect(view.tangram.addLayer).toHaveBeenCalledWith('something');
});
```

What do you think?